src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -172,8 +182,8 @@ object CancellableContinuationHolder {
 
 fun storeCancellableContinuation(cont: CancellableContinuation<*>) {
     val t = Thread.currentThread()
-    if (t is FixedActiveThreadsExecutor.TestThread) {
-        t.cont = cont
+    if (t is TestThread) {
+        t.suspendedContinuation = cont
     } else {
         CancellableContinuationHolder.storedLastCancellableCont = cont
```

Comment:
Please add a comment explaining that you need this logic for the verification phase only.